Plugins/Traccc/include/Acts/Plugins/Traccc/MeasurementConversion.hpp

```diff
@@ -0,0 +1,164 @@
+      return Acts::BoundIndices::eBoundTime;
+    case traccc::bound_indices::e_bound_size:
+      return Acts::BoundIndices::eBoundSize;
+    default:
+      throw std::runtime_error("Could not convert traccc bound index");
+  }
+}
+
+/// @brief Creates an Acts measurement from a traccc measurement.
+/// @tparam the dimension of the Acts measurement (subspace size).
+/// @param m the traccc measurement.
+/// @param sl the Acts source link to use for the Acts measurement.
+/// @returns an Acts measurement with data copied from the traccc measurement
+/// and with its source link set to the one provided to the function.
+template <std::size_t dim>
+inline Acts::Measurement<Acts::BoundIndices, dim> measurement(
+    const traccc::measurement& m, const Acts::SourceLink sl) {
+  auto params = Detail::newVector<dim>(m.local);
+  std::array<Acts::BoundIndices, dim> indices;
+  for (unsigned int i = 0; i < dim; i++) {
+    indices[i] = boundIndex(traccc::bound_indices(m.subs.get_indices()[i]));
+  }
+  auto cov = Eigen::DiagonalMatrix<Acts::ActsScalar, static_cast<int>(dim)>(
+                 Detail::newVector<dim>(m.variance))
+                 .toDenseMatrix();
+  return Acts::Measurement<Acts::BoundIndices, dim>(std::move(sl), indices,
+                                                    params, cov);
+}
+
+/// @brief Creates an Acts bound variant measurement from a traccc measurement.
+/// Using recursion, the functions determines the dimension of the traccc
+/// measurement which is used for the Acts measurement that the bound variant
+/// measurement holds. The dimension must lie between [0; max_dim].
+/// @tparam max_dim the largest possible dimension of any measurement type in the variant (default = 4).
+/// @param m the traccc measurement.
+/// @param sl the Acts source link to use for the Acts measurement.
+/// @returns an Acts bound variant measurement with data copied from the traccc measurement
+/// and with its source link set to the one provided to the function.
+template <std::size_t max_dim = 4UL>
+inline Acts::BoundVariantMeasurement boundVariantMeasurement(
+    const traccc::measurement& m, const Acts::SourceLink sl) {
+  if constexpr (max_dim == 0UL) {
+    std::string errorMsg = "Invalid/mismatching measurement dimension: " +
+                           std::to_string(m.meas_dim);
+    throw std::runtime_error(errorMsg.c_str());
+  } else {
+    if (m.meas_dim == max_dim) {
+      return measurement<max_dim>(m, sl);
+    }
+    return boundVariantMeasurement<max_dim - 1>(m, sl);
+  }
+}
+
+/// @brief Gets the the local position of the measurement.
+/// @param measurement the Acts measurement.
+/// @returns A two-dimensional vector containing the local position.
+/// The first item in the vector is local position on axis 0 and
+/// I.e., [local position (axis 0), local position (axis 1)].
+template <std::size_t dim>
+inline Acts::ActsVector<2> getLocal(
+    const Acts::Measurement<Acts::BoundIndices, dim>& measurement) {
```

Comment:
There is already a dispatch meachanism for this, which uses `std::variant` and `std::visit`. But like in the other comment, this usage of the EDM should probably be reconsidered.